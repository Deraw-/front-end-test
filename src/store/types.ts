// tslint:disable-next-line:no-empty-interface
export interface RootState {}

export interface Product extends Object {
	id: number;
	quantity: number;
}

export interface CartProduct extends Product {
	name: string;
	price: number;
	image: string;
}

// This is used to properly type components using mappers from Vuex
// see https://github.com/vuejs/vuex/pull/1121
// tslint:disable-next-line: no-empty-interface
export default interface VuexBindings {
	cartProducts: CartProduct[];
}
