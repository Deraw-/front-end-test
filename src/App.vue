<template>
	<div
		v-cloak
		class="main-ctn"
	>
		<Header />

		<main role="main">
			<!-- Main title -->
			<h1 class="sr-only">
				PÃ¢tisserie shop
			</h1>

			<!-- Load the component defined in the router for the current route -->
			<transition
				name="fade"
				mode="out-in"
			>
				<RouterView />
			</transition>
		</main>

		<Footer />
	</div>
</template>

<script lang="ts">
	import Vue from 'vue';

	import { MetaInfo } from 'vue-meta';

	import Header from '@/components/Header.vue';
	import Footer from '@/components/Footer.vue';

	export default Vue.extend({
		name: 'App',
		components: {
			Header,
			Footer
		},
		metaInfo(): MetaInfo {
			return {
				// If no subcomponents specify a metaInfo.title, this title will be used
				title: this.$t('common.meta.title') as string,
				// All titles will be injected into this template
				titleTemplate: '%s | Pastry',
				meta: [
					{
						name: 'description',
						content: this.$t('common.meta.description') as string
					}
				]
			};
		}
	});
</script>

<style lang="scss">
	// Default styles and helper classes
	* {
		$transparent: rgba(0, 0, 0, 0);

		max-width: 100%;
		margin: 0;
		padding: 0;
		border: 0;
		scroll-behavior: smooth;
		font-family: 'Roboto', sans-serif;
		box-sizing: border-box;
		word-break: break-word;
		hyphens: auto;
		-moz-osx-font-smoothing: grayscale;
		-webkit-tap-highlight-color: $transparent;
	}

	[v-cloak] {
		display: none;
	}

	html,
	body {
		// Font size defined by the browser
		font-size: 100%;
	}

	html {
		overflow-y: scroll;
	}

	body {
		min-height: 100vh;
		line-height: 1.25;
	}

	main {
		flex: 1 0 auto;

		// Space for Header
		padding-top: 64px;
	}

	input {
		appearance: none;
	}

	ul {
		list-style: none;
	}

	a,
	input,
	button {
		cursor: pointer;
		transition: .25s;
	}

	a,
	button {
		border: 1px solid transparent;

		&:focus {
			outline: none;

			border-color: hsla(0, 0%, 100%, .5);
		}
	}

	.btn {
		padding: 5px 15px;
		border-radius: 2px;

		font-size: 1rem;

		color: inherit;

		display: flex;
		align-items: center;

		&.shadow {
			&.persitent,
			&:hover,
			&:focus {
				box-shadow:
				0 3px 1px -2px rgba(0, 0, 0, .2),
				0 2px 2px 0 rgba(0, 0, 0, .14),
				0 1px 5px 0 rgba(0, 0, 0, .12);
			}
		}

		&.primary {
			display: inline-flex;

			margin: 0 -16px;
			margin-top: 24px;
			text-decoration: none;

			color: #007c8e;

			&:hover,
			&:focus {
				background: rgba(74, 186, 208, .1);
			}
		}
	}

	.sr-only {
		position: absolute !important;
		top: 0;
		left: -999px;
		width: 1px !important;
		height: 1px !important;
		white-space: nowrap;
		overflow: hidden;
		clip: rect(1px, 1px, 1px, 1px);
	}

	.required label::after {
		top: -2px;
		content: '*';
		position: relative;
	}

	// Fade <transition>
	.fade-enter-active,
	.fade-leave-active {
		transition-duration: .15s;
		transition-property: opacity;
		transition-timing-function: ease;
	}

	.fade-enter,
	.fade-leave-active {
		opacity: 0;
	}

	.main-ctn,
	main {
		width: 100%;
		display: flex;
		flex-direction: column;
	}

	main {
		align-items: center;
	}

	.page {
		margin: 0 auto;
		padding: 48px;

		display: flex;
		flex: 1 0 auto;
		flex-direction: column;

		width: 1400px;
		min-height: 500px;
	}

	.title {
		font-size: 1.75rem;
	}

	.description {
		margin-top: 15px;
		font-size: 1.1rem;
	}

	.grid-list {
		display: grid;
		grid-gap: 32px;
		grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
		grid-auto-rows: minmax(250px, auto);
		margin-top: 50px;
	}

	@media only screen and (max-width: 600px) {
		.grid-list {
			margin-top: 25px;
		}

		.page {
			padding: 15px !important;
		}
	}
</style>
