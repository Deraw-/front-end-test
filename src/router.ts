import Vue from 'vue';
import Router from 'vue-router';
import Meta from 'vue-meta';

import i18n, { loadLanguageAsync } from '@/i18n';

// Directly import Anomalies view for faster rendering of first page
import Home from '@/views/Home.vue';

Vue.use(Router);
Vue.use(Meta);

// See https://router.vuejs.org/fr/essentials/navigation.html for help
const router = new Router({
	mode: 'history', // Use the History API, and remove the hash from the URL
	routes: [
		{
			path: '/',
			name: 'home',
			component: Home
		},
		{
			path: '/cart',
			name: 'cart',
			component: () => import(/* webpackChunkName: "cart" */ './views/Cart.vue')
		},
		{
			path: '*', // Everything that doesn't match something
			name: 'notFound',
			component: () => import(/* webpackChunkName: "not-found" */ './views/NotFound.vue')
		}
	],
	// Native-like scroll behavior
	scrollBehavior(to, from, savedPosition) {
		// If there is an hash, this is an anchor, so scroll to it
		if (to.hash) {
			return {
				selector: to.hash
			};
		} else {
			// Else simulate browser behavior by returning to the previous position, or fallback to top
			if (savedPosition) {
				return savedPosition;
			} else {
				return { x: 0, y: 0 };
			}
		}
	}
});

// Resolve translations before each navigation
router.beforeEach((to, from, next) => {
	const lang = i18n.locale;
	loadLanguageAsync(lang).then(() => next());
});

export default router;
