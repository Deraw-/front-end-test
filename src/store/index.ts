import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import VuexPersistence from 'vuex-persist';
import { RootState } from './types';

// Modules
import cart from './modules/cart';
import products from './modules/products';

// The store is saved in the browser's session
const vuexLocal = new VuexPersistence({
	storage: window.sessionStorage
});

Vue.use(Vuex);

// See https://vuex.vuejs.org/fr/getting-started.html for help
const store: StoreOptions<RootState> = {
	strict: true,
	state: {},
	// See https://vuex.vuejs.org/guide/modules.html for more info on modules
	modules: {
		cart,
		products
	},
	plugins: [
		vuexLocal.plugin
	]
};

export default new Vuex.Store<RootState>(store);
