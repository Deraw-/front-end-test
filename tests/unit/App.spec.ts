import { shallowMount } from '@vue/test-utils';
import App from '@/App.vue';

import localVue from 'tests-unit/index';
import router from 'tests-unit/router';
import store from '@/store/';

// Build component before each test
let wrapper: any;
beforeEach(() => {
	wrapper = shallowMount(App, {
		localVue,
		store,
		router
	});
});

// Tests
describe('App.vue', () => {
	it('renders correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
