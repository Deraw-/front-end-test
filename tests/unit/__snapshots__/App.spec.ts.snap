// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders correctly 1`] = `
<div class="main-ctn">
  <header-stub></header-stub>
  <main role="main">
    <h1 class="sr-only">
      PÃ¢tisserie shop
    </h1>
    <routerview-stub name="fade" mode="out-in"></routerview-stub>
  </main>
  <footer-stub></footer-stub>
</div>
`;
